velox/exec/TableWriter.cpp

```diff
@@ -115,6 +120,44 @@ RowVectorPtr TableWriter::getOutput() {
 
   std::vector<std::string> fragments = dataSink_->finish();
 
+  if (!aggregation_->isFinished()) {
+    vector_size_t numOutputRows = 1;
+    std::vector<VectorPtr> columns;
+    auto aggregationOutput = aggregation_->getOutput();
+    auto commitContextJson = folly::toJson(folly::dynamic::object(
+        "lifespan", "TaskWide")("taskId", connectorQueryCtx_->taskId())(
+        "pageSinkCommitStrategy", commitStrategyToString(commitStrategy_))(
+        "lastPage", false));
+    // clang-format on
+
+    auto commitContextVector = std::make_shared<ConstantVector<StringView>>(
+        pool(),
+        numOutputRows,
+        false /*isNull*/,
+        VARBINARY(),
+        StringView(commitContextJson));
+
+    for (int channel = 0; channel < outputType_->size(); channel++) {
+      if (channel < CONTEXT_CHANNEL) {
+        auto row_count_channel_or_fragment = BaseVector::create(
+            outputType_->childAt(channel),
+            aggregationOutput->childAt(0)->size(),
+            pool());
+        row_count_channel_or_fragment->setNull(0, true);
+        columns.push_back(row_count_channel_or_fragment);
+      } else if (channel == CONTEXT_CHANNEL) {
+        columns.push_back(commitContextVector);
+      } else {
+        columns.push_back(
+            aggregationOutput->childAt(channel - STATS_START_CHANNEL));
```

Comment:
You can have a utility to do the type conversion based on the the actual type in aggregates.